keps/sig-node/3960-pod-lifecycle-sleep-action/README.md

```diff
@@ -252,6 +256,16 @@ N/A
   4. For each termination grace period value, delete the pod and observe the time it takes for the container to terminate.
   5. Verify that the container is terminated after the min(sleep, grace).
 
+Tests List
+- [pod-lifecycle-sleep-action test](https://github.com/kubernetes/kubernetes/blob/a1ffdedf782edf1472102b0b99c1467d4ed39753/test/e2e/common/node/lifecycle_hook.go#L550)
```

Comment:
Could you also link a job that runs this test? I guess in this case it would be a e2e alpha job? Just want to make sure these tests are actually run somewhere in our CI

Reply:
Yea... About this, I think it should be `pull-kubernetes-e2e-gce-cos-alpha-features` , at least it runs these tests back when I merge codes to k/k.
However, it seems that these tests were skipped in a long time. https://testgrid.k8s.io/presubmits-kubernetes-nonblocking#pull-kubernetes-e2e-gce-cos-alpha-features
I don't know why it's been skip, also not sure if I should paste it if it's not running periodically.

Reply:
So couple things, these jobs are optional and are triggered via a PR. You also have periodics jobs that run on schedule.

I kicked off test for this. It does look like your jobs ran at some time (and failed) but I did a search here:

https://storage.googleapis.com/k8s-triage/index.html?date=2024-01-24&pr=1&test=PodLifecycleSleepAction#1452fc8fa29030df31d4

I think confirming that a job covers this is enough for the beta promotion. I'd like the jobs and maybe consider verifying that it is running in a periodic job.

Reply:
I'm really not familiar with sig-testing, and rarely used Testgrid before, so I'll have to ask some questions.

First of all, the failure of these jobs is because these jobs are a bit flaky in CI. Since we compare time cost during the test, the pressure of the cluster will significantly affect the results. I'll relax the judgement in the beta pr.

The first question:
> https://storage.googleapis.com/k8s-triage/index.html?date=2024-01-24&pr=1&test=PodLifecycleSleepAction#1452fc8fa29030df31d4

Hmmm, this links shows some failures from `pull-kubernetes-e2e-kind-alpha-features`, but I checked the [Testgrid](https://testgrid.k8s.io/presubmits-kubernetes-nonblocking#pull-kubernetes-e2e-kind-alpha-features) yesterday and I double checked today, there's no PodLifecycleSleepAction test in the list of Testgrid.  
So this does means that this test is **not** ran periodically but are triggered via some PR? 
And if so, do I need to make one job to periodically ran this test? I think I don't need to as we're going to promote this to beta and there should be some jobs periodically run tests for beta features.

The second question:
This question is actually not related to this pr, but k/k😅. 
How do we handle the e2e test when we promote one feature to beta? 
Use this test as an example:
Do we remove the `[Feature:PodLifecycleSleepAction]` from test name`[sig-node] [Feature:PodLifecycleSleepAction] xxxx` or do we keep it as it is. I tried and seems they both work well.
And are there some jobs periodically run tests for **all** beta features? Won't `pull-kubernetes-e2e-kind-beta-features` behaves the same as `pull-kubernetes-e2e-kind-alpha-features` and just skips this test unless it's triggered via some PR?
@kannon92 
/cc @thockin

Reply:
Came back with answers, after asked folks from sig-testing.
https://kubernetes.slack.com/archives/C09QZ4DQB/p1706249714510469
Yes, there are currently no job running tests for features periodically. Only triggered in `pull-kubernetes-e2e-gce-cos-alpha-features`

I'll try to make these tests run periodically during beta, currently there seems to be 2 options:

1. Adding this feature gate to https://github.com/kubernetes/test-infra/blob/9099855a2873a7dfc76afbe61c3e6108e4b1fed2/config/jobs/kubernetes/sig-cloud-provider/gcp/gcp-gce.yaml#L412 and `pull-kubernetes-e2e-gce-cos-alpha-features` will run this test every time.

1. Remove `[Feature:PodLifecycleSleepAction]` from test name, and it will run periodically like a normal test.

I personally prefer option 2, if it does no harm.

Reply:
I think option 1 is better, but this is not my expertise.  Deadline is looming.

Reply:
Yea its a bit of a chicken and the egg problem..

I think 1 is the right approach. If there is push back and you can't promote in 1.30, at least we have coverage for the e2e tests so we know if things break.

Reply:
Then let's go with option 1, I added test-grid link of `pull-kubernetes-e2e-gce-cos-alpha-features` to KEP document. After beta code merged, we'll see results in this grid.
@kannon92 @thockin If everything else is fine. Can anyone of you give me a `lgtm`? Seems that we need to find other approvers to approve this KEP.